engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/PreviewWorldScreen.java

```diff
@@ -115,13 +135,63 @@ public void onOpened() {
     }
 
     @Override
+    public void update(float delta) {
+         super.update(delta);
+
+         if (triggerUpdate) {
+             updatePreview();
+             triggerUpdate = false;
+         }
+    }
+
+    private void configureProperties() {
+
+        PropertyLayout properties = find("properties", PropertyLayout.class);
+        properties.setOrdering(PropertyOrdering.byLabel());
+        properties.clear();
+
+        WorldConfigurator worldConfig = worldGenerator.getConfigurator();
+
+        Map<String, Component> params = worldConfig.getProperties();
+
+        for (String key : params.keySet()) {
+            Class<? extends Component> clazz = params.get(key).getClass();
+            Component comp = config.getModuleConfig(worldGenerator.getUri(), key, clazz);
+            if (comp != null) {
+                worldConfig.setProperty(key, comp);       // use the data from the config instead of defaults
+            }
+        }
+
+        for (String label : params.keySet()) {
+            PropertyProvider<?> provider = new PropertyProvider<>(params.get(label));
+            properties.addPropertyProvider(label, provider);
+        }
+    }
+
+    @Override
     public void onClosed() {
-        super.onClosed();
+
+        CoreRegistry.remove(WorldGeneratorPluginLibrary.class);
```

Comment:
On a local branch of mine I have already removed the remove method of CoreRegistry.

That you remove it here is a sign that it shouldn't be in the CoreRegistry for the first place.

If you need a Context that contains objects only temporarly then you can create a child context via ContextImpl(context) that implicitly contains all objects of the parent. You can then add your temporary objects to the child context and use it wherever you need a context with that temporary object. Once you are done you can simply throw away the child context and the parent context remains unchanged.